recipes/botan/all/conanfile.py

```diff
@@ -249,11 +249,16 @@ def _configure_cmd(self):
         if self.settings.os != "Windows" and self.options.fPIC:
             botan_extra_cxx_flags.append('-fPIC')
 
-        if self.settings.os == "Macos" and self.settings.os.version:
-            macos_min_version = tools.apple_deployment_target_flag(self.settings.os,
-                                                                   self.settings.os.version)
+        if tools.is_apple_os(self.settings.os):
+            if self.settings.get_safe("os.version"):
```

Comment:
this one is the correct form, checking other recipes

Reply:
Sorry my question did not get posted (human error) 

How does the get safe know the recursion?

OS is always present, why is that the preference?

Reply:
I don't think it's always present, you always may have custom settings.yml without os. anyway, most of the recipes use `settings.get_safe` for nested things like `compiler.runtime` or `os.version`, so I'd stick with it to be consistent.

Reply:
(your suggestion has been posted, and I've applied it, but then realized it looks awkward, so I've reverted back)